opm/simulators/linalg/bda/opencl/Preconditioner.cpp

```diff
@@ -40,13 +40,15 @@ void Preconditioner<block_size>::setOpencl(std::shared_ptr<cl::Context>& context
 }
 
 template <unsigned int block_size>
-std::unique_ptr<Preconditioner<block_size> > Preconditioner<block_size>::create(PreconditionerType type, int verbosity, bool opencl_ilu_parallel) {
-    if (type == PreconditionerType::BILU0) {
-        return std::make_unique<Opm::Accelerator::BILU0<block_size> >(opencl_ilu_parallel, verbosity);
-    } else if (type == PreconditionerType::CPR) {
-        return std::make_unique<Opm::Accelerator::CPR<block_size> >(verbosity, opencl_ilu_parallel);
-    } else if (type == PreconditionerType::BISAI) {
-        return std::make_unique<Opm::Accelerator::BISAI<block_size> >(opencl_ilu_parallel, verbosity);
+std::unique_ptr<Preconditioner<block_size>>
+Preconditioner<block_size>::create(Type type, int verbosity, bool opencl_ilu_parallel)
+{
+    if (type == Type::BILU0) {
+        return std::make_unique<BILU0<block_size> >(opencl_ilu_parallel, verbosity);
+    } else if (type == Type::CPR) {
+        return std::make_unique<CPR<block_size> >(verbosity, opencl_ilu_parallel);
+    } else if (type == Type::BISAI) {
+        return std::make_unique<BISAI<block_size> >(opencl_ilu_parallel, verbosity);
     } else {
         OPM_THROW(std::logic_error, "Invalid PreconditionerType");
     }
```

Comment:
Would it make sense to replace this with a `switch` instead?  That way, the compiler will warn us if we forget one of the alternatives when introducing a new enumerator.  E.g.,
```C++
switch (type) {
case Type::BILU0:
    return ...
    
case Type::CPR:
    return ...

case Type::BISAI:
    return ...
}

OPM_THROW(std::logic_error, "Unsupported preconditioner type " + std::to_string(static_cast<int>(type)));
```

Reply:
I feared i would fall in that rabbit hole haha. Working in this code I'm constantly fighting the urge to inflate the commits with all sorts of cosmetics and formatting things. Anyways since you called these out; yes the switch makes sense, done. The inconsistent order is indeed inconsistent. I've take care of that as well.

Reply:
> Working in this code I'm constantly fighting the urge to inflate the commits with all sorts of cosmetics and formatting things.

Right.  If you're doing multiple steps, then we can hold off on those kinds of things until everything else has landed.  Just let me know and then I'll not make those kinds of comments.

Reply:
Oh, please do make suggestion, it only makes the code better.. Don't read it as annoyance with those. It's just the BDA code is really out of wack when it comes to formatting and general little niggles. Too few eyes..